src/overlays/actors/ovl_En_Tite/z_en_tite.c

```diff
@@ -39,55 +103,877 @@ const ActorInit En_Tite_InitVars = {
+            if (this->actor.groundY > BGCHECK_Y_MIN) {
+                this->actor.posRot.pos.y = this->actor.groundY;
+            }
+        } else {
+            this->actor.velocity.y = 0.0f;
+            this->actor.gravity = 0.0f;
+            this->actor.posRot.pos.y += this->actor.yDistToWater;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1A6E4.s")
+    // play sound and generate ripples
+    if (this->actor.bgCheckFlags & 0x42) {
+        if (!(this->actor.bgCheckFlags & 0x40)) {
+            func_80033480(globalCtx, &this->frontLeftFootPos, 1.0f, 2, 80, 15, 1);
+            func_80033480(globalCtx, &this->frontRightFootPos, 1.0f, 2, 80, 15, 1);
+            func_80033480(globalCtx, &this->backRightFootPos, 1.0f, 2, 80, 15, 1);
+            func_80033480(globalCtx, &this->backLeftFootPos, 1.0f, 2, 80, 15, 1);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+        } else {
+            this->actor.bgCheckFlags = (this->actor.bgCheckFlags & ~0x40);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_TEKU_LAND_WATER2);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1A76C.s")
+    // If player is far away, idle. Otherwise attack or move
+    angleToPlayer = (this->actor.yawTowardsLink - this->actor.shape.rot.y);
+    if ((this->actor.speedXZ == 0.0f) &&
+        ((this->actor.bgCheckFlags & 1) || (TEKTITE_BLUE && (this->actor.bgCheckFlags & 0x20)))) {
+        this->actor.posRot.rot.y = this->actor.shape.rot.y;
+        this->collider.base.atFlags &= ~0x2;
+        if ((this->actor.xzDistToLink > 300.0f) && (this->actor.yDistToLink > 80.0f) &&
+            (ABS(this->actor.shape.rot.x) < 4000) && (ABS(this->actor.shape.rot.z) < 4000) &&
+            ((this->actor.bgCheckFlags & 1) || (TEKTITE_BLUE && (this->actor.bgCheckFlags & 0x20)))) {
+            EnTite_SetupIdle(this);
+        } else if ((this->actor.xzDistToLink < 180.0f) && (this->actor.yDistToLink <= 80.0f) &&
+                   (ABS(angleToPlayer) <= 6000)) {
+            EnTite_SetupAttack(this);
+        } else {
+            EnTite_SetupMoveTowardPlayer(this);
+        }
+    }
+    SkelAnime_Update(&this->skelAnime);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1A888.s")
+void EnTite_SetupStunned(EnTite* this) {
+    Animation_Change(&this->skelAnime, &D_060012E4, 0.0f, 0.0f, (f32)Animation_GetLastFrame(&D_060012E4), 0, 4.0f);
+    this->action = TEKTITE_STUNNED;
+    this->actor.speedXZ = -6.0f;
+    this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+    if (this->damageEffect == 0xF) {
+        this->spawnIceTimer = 48;
+    }
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+    EnTite_SetupAction(this, EnTite_Stunned);
+}
+
+/**
+ * stunned or frozen
+ */
+void EnTite_Stunned(EnTite* this, GlobalContext* globalCtx) {
+    s16 angleToPlayer;
```

Comment:
It's not very important but we generally try to leave a line between declarations and the rest
```suggestion
    s16 angleToPlayer;

```